compiler/optimizer/Inliner.cpp

```diff
@@ -1949,7 +1949,8 @@ TR_InlinerBase::addGuardForVirtual(
        (guard->_kind == TR_HierarchyGuard))
       shouldAttemptOSR = false;
 
-   if (callerSymbol->supportsInduceOSR(callNode->getByteCodeInfo(), block1, calleeSymbol, comp()))
+   if (comp()->getHCRMode() != TR::osr
```

Comment:
Ditto re: my earlier comment, knowing what the "osr" HCR mode here would help but perhaps so would a comment in the code on why this logic checks for OSR induction only if HCR mode is not "osr".